src/industry_cmd.cpp

```diff
@@ -2689,6 +2713,10 @@ static void ChangeIndustryProduction(Industry *i, bool monthly)
 					new_prod = Clamp(new_prod, 0, 16);
 				}
 
+				/* If override flags are set, prevent actually changing production if any was decided on */
+				if ((i->ctlflags & INDCTL_NO_PRODUCTION_DECREASE) && new_prod < old_prod) continue;
+				if ((i->ctlflags & INDCTL_NO_PRODUCTION_INCREASE) && new_prod > old_prod) continue;
+
```

Comment:
The code below this, related to `closeit = false;` has to be placed before this block. Otherwise, industries will be announcing imminent closures the next month should a production change occur.

I re-arranged it like this:
```				/* Do not stop closing the industry when it has the lowest possible production rate */
				if (new_prod == old_prod && old_prod > 1) {
					closeit = false;
					continue;
				}

				/* Close the industry when it has the lowest possible production rate */
				if (new_prod > 1) closeit = false;

				/* If override flags are set, prevent actually changing production if any was decided on */
				if ((i->ctlflags & INDCTL_NO_PRODUCTION_DECREASE) && new_prod < old_prod) continue;
				if ((i->ctlflags & INDCTL_NO_PRODUCTION_INCREASE) && new_prod > old_prod) continue;

				percent = (old_prod == 0) ? 100 : (new_prod * 100 / old_prod - 100);
				i->production_rate[j] = new_prod;
```
... and it seems to work after testing.

Reply:
Fixed this in a different way.